### PR TITLE
multi swab support

### PR DESCRIPTION
**Pull request by zdeveloper**
## Related Issue or Background Info

- part 2 of #2536

## Changes Proposed

- added multi-select for swab types

## Screenshots / Demos
image

image

### UI
- [x] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/resources/graphql/main.graphqls [hunk 1]
```diff
@@ -59,9 +59,18 @@ type DeviceType {
   model: String
   loincCode: String
   swabType: String
+  swabTypes: String
```
Comment (zdeveloper) [inline backend/src/main/resources/graphql/main.graphqls:62]: this was added due to the following error
`[{"message":"Validation error of type FieldUndefined: Field 'swabTypes' in type 'DeviceType' is undefined @ 'createDeviceTypeNew/swabTypes'","locations":[]}]`

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/DeviceType.java [hunk 2]
```diff
@@ -26,6 +27,8 @@
   @Column(nullable = false)
   private String swabType;
 
+  @Transient private String swabTypes;
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/DeviceType.java:30]: added this to satisfy graphql type, this is temporary and will be used when we fully support multiple swab types

**Comment by kenieh** [comment]
LGTM!

**Comment by nathancrtr** [comment]
How do we want to handle validation for this form? This won't be widely used, which takes some urgency off, but there's a _lot_ of goofy stuff that it currently allows.

**Comment by zdeveloper** [comment]
> How do we want to handle validation for this form? This won't be widely used, which takes some urgency off, but there's a _lot_ of goofy stuff that it currently allows.

That's definitely something to think about, I dont think its urgent since its internal screen only for super admins, certainly not part of this ticket, but i would defer to @tim-skylight and @nullflux

File: frontend/src/app/commonComponents/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  useEffect(() => {
+    const sortedOptions = getSortedOptions([...options]);
+    setAvailableOptions(sortedOptions);
+  }, [options, setAvailableOptions]);
+
+  useEffect(() => {
+    if (selectedItems) {
+      onChange(selectedItems);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (nullflux) [inline frontend/src/app/commonComponents/MultiSelect/MultiSelect.tsx:106]: **suggestion:** Please add a comment explaining why `onChange` can't be in the dep list

File: frontend/src/app/patients/Components/PersonForm.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React, { useCallback, useState, useEffect, useRef } from "react";
 import { Prompt } from "react-router-dom";
 import { SchemaOf } from "yup";
 import { useTranslation } from "react-i18next";
+import { ComboBox } from "@trussworks/react-uswds";
```
Comment (nullflux) [inline frontend/src/app/patients/Components/PersonForm.tsx:5]: **praise:** Thank you for this cleanup!

**Review by nullflux — APPROVED** [review]
LGTM! Couple of minor suggestions :)

**Comment by nullflux** [comment]
> That's definitely something to think about, I dont think its urgent since its internal screen only for super admins, certainly not part of this ticket, but i would defer to @tim-skylight and @nullflux

I fall into the "validate this later" camp – let's check w/ Tim N during planning

**Pull request merged**